service/routes.py

```diff
@@ -208,6 +208,41 @@ def update_rating_of_product(product_id):
     return jsonify(product.serialize()), status.HTTP_200_OK
 
 
+######################################################################
+# UPDATE THE PRICE OF A PRODUCT
+######################################################################
+@app.route("/products/<int:product_id>/price", methods=["PUT"])
+def update_price_of_product(product_id):
+    """
+    Updates the price of a product on the basis of feedback provided.
+    Args:
+        product_id (_type_): _description_
+    """
+    app.logger.info("Request to update the price of the product with id: %s", product_id)
+    check_content_type("application/json")
+    product = Product.find(product_id)
+    if not product:
+        app.logger.info("Product_id not found.")
+        abort(
+            status.HTTP_404_NOT_FOUND, description=f"Product with id '{product_id}' was not found."
+        )
+    new_price = request.get_json()
+    if not isinstance(new_price["price"], float) and not isinstance(new_price["price"], int):
+        abort(
+            status.HTTP_406_NOT_ACCEPTABLE, description="Price should be of float or int datatype"
+        )
+    new_price["price"] = float(new_price["price"])
```

Comment:
The Condition : `if new_price["price"] is not None:` should come before the call `new_price["price"] = float(new_price["price"])`. Otherwise, it might throw a Server Error if someone sends `"price" = null`.